flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/join/TemporalJoinTest.scala

```diff
@@ -508,6 +508,27 @@ class TemporalJoinTest extends TableTestBase {
         " table, but the rowtime types are TIMESTAMP_LTZ(3) *ROWTIME* and TIMESTAMP(3) *ROWTIME*.",
       classOf[ValidationException]
     )
+
+    val sqlQuery9 = "SELECT * " +
+      "FROM Orders AS o JOIN " +
+      "RatesHistoryWithPK FOR SYSTEM_TIME AS OF 'o.rowtime' AS r " +
+      "ON o.currency = r.currency"
+    expectExceptionThrown(
+      sqlQuery9,
+      "The system time period specification expects Timestamp type but is 'CHAR'",
```

Comment:
nit: the error message would be better if it was changed `...expects Timestamp type but is...` --> `...expects Timestamp type, but got ...`

Reply:
Unforunately, this message is actually defined in Calcite. We'd need to change it there.

Reply:
IIRC this message is defined in resource file in Calcite and we (downstream project) can redefine it 
However i guess it is out of scope for this PR